src/backend/winit.rs

```diff
@@ -692,170 +701,181 @@ impl InputBackend for WinitInputBackend {
-                    let time = ((1000 * duration.as_secs()) + (nanos / 1_000_000)) as u32;
-                    match (event, handler.as_mut(), events_handler.as_mut()) {
-                        (WindowEvent::Resized(size), _, events_handler) => {
-                            trace!(logger, "Resizing window to {:?}", size);
-                            window.window().set_inner_size(size);
-                            let mut wsize = window_size.borrow_mut();
-                            wsize.logical_size = size;
-                            let physical_size = size.to_physical(wsize.dpi_factor);
-                            if let Window::Wayland { ref surface, .. } = **window {
-                                surface.resize(physical_size.width as i32, physical_size.height as i32, 0, 0);
-                            }
-                            if let Some(events_handler) = events_handler {
-                                events_handler.resized(physical_size.into(), wsize.dpi_factor);
-                            }
-                        }
-                        (WindowEvent::Focused(focus), _, Some(events_handler)) => {
-                            events_handler.focus_changed(focus)
+            
+            self.events_loop.run_return(move |event, _target, control_flow| {
+                match event {
+                    Event::RedrawEventsCleared => {
+                        *control_flow = ControlFlow::Exit;
+                    },
+                    Event::RedrawRequested(_id) => {
+                        if let Some(events_handler) = events_handler.as_mut() {
+                            events_handler.refresh();
                         }
-                        (WindowEvent::Refresh, _, Some(events_handler)) => events_handler.refresh(),
-                        (WindowEvent::HiDpiFactorChanged(factor), _, events_handler) => {
-                            let mut wsize = window_size.borrow_mut();
-                            wsize.dpi_factor = factor;
-                            let physical_size = wsize.logical_size.to_physical(factor);
-                            if let Window::Wayland { ref surface, .. } = **window {
-                                surface.resize(physical_size.width as i32, physical_size.height as i32, 0, 0);
+                    },
+                    Event::WindowEvent { event, .. } => {
+                        let duration = Instant::now().duration_since(*time);
+                        let nanos = duration.subsec_nanos() as u64;
+                        let time = ((1000 * duration.as_secs()) + (nanos / 1_000_000)) as u32;
+                        match (event, handler.as_mut(), events_handler.as_mut()) {
+                            (WindowEvent::Resized(psize), _, events_handler) => {
+                                trace!(logger, "Resizing window to {:?}", psize);
+                                window.window().set_inner_size(psize);
```

Comment:
IIRC when we receive a `Resized` event from winit, it now means that the window has been resized, so I think this line should be removed. It might be worth double checking though.

Reply:
Done.